docs/development/getting-started.rst

```diff
@@ -44,6 +63,27 @@ You can also specify a subset of tests to run as positional arguments:
     $ # run the whole x509 testsuite, plus the fernet tests
     $ nox -e tests -p py310 -- tests/x509/ tests/test_fernet.py
 
+For TDD you might want to speed up the test run cycle.
+After activating your virtualenv, install the dev requirements
+and then install the project in edit mode.
+
+.. code-block:: console
+
+    $ pip install -e vectors/
+    $ pip install -e .[ssh,test,test-randomorder]
+
+If you are changing the OpenSSL bindings from `src/_cffi_src` you will need
+to rebuild the rust OpenSSL bindings.
+The easiest method is to just reinstall in edit mode.
+
+.. code-block:: console
+
+    $ # Re-build the binding and other extensions.
+    $ pip install -e .
+    $ # Then re-run the tests.
+    $ pytest tests/hazmat/bindings/
+
+
```

Comment:
I'm opposed to documenting this. While it works (and it's how many core devs develop), it's far too easy to screw up: notably failing to rebuild any extension module will get you confusing and inconsistent results, that's why our documentation is about `nox`.

Reply:
Maybe add a warning before this section ?

Reply:
Unless @reaperhulk disagrees with me, I don't want to add it.

Perhaps we can suggest `nox -e tests-nocoverage` (which reduces some time), or the usage of `--no-install`, which can be used on subsequent invocations if you haven't changed any extension modules.

Reply:
No problem.

I now know how to do TDD for crypography.
I was thinking that this info is useful for others.

-------

Disabling `-n auto` can also save 6 seconds.

Reply:
That's interesting that disable parallelism saves time, what sort of CPU do you have? I almost always run the full test suite, and it's a big time savings for me (10 cores; 2 of which are efficiency cores).

Reply:
This is only for TDD... when you run a subset of the tests

```
$ time pytest -n auto --dist=worksteal --cov=cryptography --cov=tests --durations=10 tests/hazmat/bindings/test_openssl.py

real    0m6.638s
user    0m27.316s
sys     0m3.032s

vs

$ time pytest tests/hazmat/bindings/test_openssl.py

real    0m0.587s
user    0m0.522s
sys     0m0.070s
```